File: ejercicioscapitulo2/volumencilindro.c
/* fichero volumencilindro.c */
/* Este programa calcula el volumen de un cilindro*/
#include <stdio.h>
#include <stdlib.h>

#define PI_CONST 3.141592653589793

int main(void) {
    
    double D, H, R, V;
    int valor_leido;

    printf ("Introduzca el diámetro, en metros: "); 
    valor_leido = scanf ("%lf", &D);
    
    if (valor_leido != 1 || D <= 0.0) {
        printf("Error: El diámetro debe ser un número positivo.\n");
        return EXIT_FAILURE;
    }
    
    printf ("Introduzca la altura, en metros: "); 
    valor_leido = scanf ("%lf", &H);

    if (valor_leido != 1 || H <= 0.0) {
        printf("Error: La altura debe ser un número positivo.\n");
        return EXIT_FAILURE;
    }
    
    R = D / 2.0; 
    V = PI_CONST * R * R * H; 
    printf ("El volumen del cilindro es de %.4lf metros cúbicos.\n", V);
    
    return EXIT_SUCCESS;
}

File: ejercicioscapitulo2/VolumenCilindro.java
// fichero VolumenCilindro.java
// Este programa calcula el volumen de un cilindro
import java.util.Scanner;
import java.util.InputMismatchException;

public class VolumenCilindro {

  public static void main(String[] args) {
        
    try (Scanner scanner = new Scanner(System.in)) {
      
      System.out.print("Introduzca el diámetro, en metros: ");
      double diametro = scanner.nextDouble(); 
      System.out.print("Introduzca la altura, en metros: ");
      double altura = scanner.nextDouble(); 
      
      if (diametro <= 0 || altura <= 0) {
          System.err.println("Error: El diámetro y la altura deben ser valores positivos.");
          return;
      }
      
      double radio = diametro / 2.0;    
      double volumen = Math.PI * Math.pow(radio, 2) * altura;
      System.out.printf("El volumen del cilindro es de %.4f metros cúbicos%n", volumen);

    } catch (InputMismatchException e) {
        System.err.println("Error: Por favor, introduzca valores numéricos válidos.");
      
    } catch (Exception e) {
        System.err.println("Ha ocurrido un error inesperado.");
        e.printStackTrace();
    }
  }
}